zeebe/atomix/cluster/src/test/java/io/atomix/cluster/messaging/impl/NettyMessagingServiceTest.java

```diff
@@ -204,19 +212,46 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg)
             .addBefore(
                 HeartbeatHandler.HEARTBEAT_HANDLER_NAME, "test-interceptor", interceptingHandler);
 
-        // then - payload should have content only when both client and server support
-        // it
-        Awaitility.await("Until heartbeats payloads are received")
+        // then - verify heartbeats are exchanged with correct payload based on configuration
+        Awaitility.await("Until heartbeats are received")
             .atMost(Duration.ofSeconds(2))
             .untilAsserted(
                 () -> {
                   if (clientSupportsPayload && serverSupportsPayload) {
+                    // Both support payloads: verify payloads with timestamps
                     assertThat(heartbeatFromClient.get())
                         .isNotNull()
                         .satisfies(heartbeat -> assertThat(heartbeat.sentAt()).isPositive());
                     assertThat(heartbeatResponseFromServer.get())
                         .isNotNull()
                         .satisfies(heartbeat -> assertThat(heartbeat.receivedAt()).isPositive());
+                  } else {
+                    // At least one party doesn't support payloads: verify empty heartbeats
+                    assertThat(
+                            emptyHeartbeatFromClientReceived.get()
+                                || heartbeatFromClient.get() != null)
+                        .as("Heartbeat from client should be received (empty or with payload)")
+                        .isTrue();
+                    assertThat(
+                            emptyHeartbeatResponseFromServerReceived.get()
+                                || heartbeatResponseFromServer.get() != null)
+                        .as(
+                            "Heartbeat response from server should be received (empty or with payload)")
+                        .isTrue();
+
+                    // Additionally verify that at least one side has empty payload as expected
+                    if (!clientSupportsPayload) {
```

Comment:
The payload should be empty if one of the two does not support sending payloads, these assertions feel incorrect